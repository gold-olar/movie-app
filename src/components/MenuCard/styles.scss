.menu-card {
  background: none !important;
}
.btn-primary {
  background-color: transparent !important;
  text-transform: uppercase;
  font-weight: 400 !important;
  opacity: 0.8;
  border: 1px solid #ccc !important;
}
.btn-primary:hover {
  background-color: #db0000 !important;
  border: 1px solid transparent !important;
}
.menu-card-title {
  font-size: 160%;
}
.menu-card-body{
    opacity: 0.5;
    font-weight: 300 !important;
}