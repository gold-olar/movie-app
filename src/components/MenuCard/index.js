import React from "react";
import PropTypes from "prop-types";
import { Card, CardTitle, CardText, Col } from "reactstrap";
import "./styles.scss";
import { Link } from "react-router-dom";

const MenuCard = ({ type, description, path }) => {
  return (
    <Col sm="6">
      <Card className="menu-card" body>
        <CardTitle className="menu-card-title">{type}</CardTitle>
        <CardText className="menu-card-body">{description}</CardText>
        <Link to={path} className="btn btn-primary">
          Browse {type}
        </Link>
      </Card>
    </Col>
  );
};

MenuCard.propTypes = {
  type: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};

export default MenuCard;
