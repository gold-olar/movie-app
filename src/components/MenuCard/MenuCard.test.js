import React from "react";
import { BrowserRouter } from "react-router-dom";
import { unmountComponentAtNode } from "react-dom";
import ReactDOM from "react-dom";
import MenuCard from "./index";
import renderer from "react-test-renderer";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
const testComponent = (
  <BrowserRouter>
    <MenuCard type="series" description="any string" path="/path" />{" "}
  </BrowserRouter>
);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(testComponent, div);
});

test("should match snapshot", () => {
  const component = renderer.create(testComponent);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
