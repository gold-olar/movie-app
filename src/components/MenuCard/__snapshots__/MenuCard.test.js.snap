// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  <div
    className="col-sm-6"
  >
    <div
      className="menu-card card card-body"
    >
      <div
        className="menu-card-title card-title"
      >
        series
      </div>
      <p
        className="menu-card-body card-text"
      >
        any string
      </p>
      <a
        className="btn btn-primary"
        href="/path"
        onClick={[Function]}
      >
        Browse 
        series
      </a>
    </div>
  </div>,
  " ",
]
`;
